lambdas/data-migration2/src/granulesAndFiles.ts

```diff
@@ -0,0 +1,190 @@
+  knex: Knex.Transaction
+): Promise<number> => {
+  // Validate record before processing using API model schema
+  Manager.recordIsValid(record, schemas.granule);
+  const { name, version } = deconstructCollectionId(record.collectionId);
+  const collectionPgModel = new CollectionPgModel();
+  const granulePgModel = new GranulePgModel();
+
+  const collectionCumulusId = await collectionPgModel.getRecordCumulusId(
+    knex,
+    { name, version }
+  );
+
+  const existingRecord = await granulePgModel.get(knex, {
+    granule_id: record.granuleId,
+    collection_cumulus_id: collectionCumulusId,
+  });
+
+  // Throw error if it was already migrated.
+  if (existingRecord) {
+    throw new RecordAlreadyMigrated(`Granule ${record.granuleId} was already migrated, skipping`);
+  }
+
+  const granule = await translateApiGranuleToPostgresGranule(record, knex);
+  const [cumulusId] = await granulePgModel.upsert(knex, granule);
+  return cumulusId;
+};
+
+/**
+ * Migrate File record from a Granules record from DynamoDB  to RDS.
+ *
+ * @param {ApiFile} file - Granule file
+ * @param {number} granuleCumulusId - ID of granule
+ * @param {Knex.Transaction} trx - Knex transaction
+ * @returns {Promise<void>}
+ * @throws {RecordAlreadyMigrated} if record was already migrated
+ */
+export const migrateFileRecord = async (
+  file: ApiFile,
+  granuleCumulusId: number,
+  trx: Knex.Transaction
+): Promise<void> => {
+  const filePgModel = new FilePgModel();
+
+  const bucket = getBucket(file);
+  const key = getKey(file);
+
+  // Map old record to new schema.
+  const updatedRecord: PostgresFile = {
+    bucket,
+    key,
+    granule_cumulus_id: granuleCumulusId,
+    file_size: file.size,
+    checksum_value: file.checksum,
+    checksum_type: file.checksumType,
+    file_name: file.fileName,
+    source: file.source,
+    path: file.path,
+  };
+  await filePgModel.upsert(trx, updatedRecord);
+};
+
+/**
+ * Migrate granule and files from DynamoDB to RDS
+ * @param {AWS.DynamoDB.DocumentClient.AttributeMap} dynamoRecord
+ * @param {GranulesAndFilesMigrationSummary} granuleAndFileMigrationSummary
+ * @param {Knex} knex
+ * @returns {Promise<MigrationSummary>} - Migration summary for files
+ */
+export const migrateGranuleAndFilesViaTransaction = async (
+  dynamoRecord: AWS.DynamoDB.DocumentClient.AttributeMap,
+  granuleAndFileMigrationSummary: GranulesAndFilesMigrationSummary,
+  knex: Knex
+): Promise<GranulesAndFilesMigrationSummary> => {
+  const files = dynamoRecord.files;
+  const { granulesSummary, filesSummary } = granuleAndFileMigrationSummary;
+
+  granulesSummary.dynamoRecords += 1;
+  filesSummary.dynamoRecords += files.length;
+
+  try {
+    await knex.transaction(async (trx) => {
+      const granuleCumulusId = await migrateGranuleRecord(dynamoRecord, trx);
+      await Promise.all(files.map(async (file : ApiFile) => {
+        await migrateFileRecord(file, granuleCumulusId, trx);
+      }));
```

Comment:
```suggestion
      return Promise.all(files.map(async (file : ApiFile) => migrateFileRecord(file, granuleCumulusId, trx)));
```

Reply:
I typically am in favor of condensing code to a single line but the line exceeds 100 characters and it doesn't seem feasible to rename anything in the proposed line to condense the character count

Reply:
```suggestion
      return Promise.all(files.map(
        async (file : ApiFile) => migrateFileRecord(file, granuleCumulusId, trx)
      ));
```